presto-main/src/main/java/com/facebook/presto/server/StatementResource.java

```diff
@@ -450,6 +450,14 @@ else if (!queryInfo.getOutputStage().isPresent()) {
                 }
             }
 
+            List<QueryError> warnings;
+            if (queryInfo.getState().isDone()) {
+                warnings = queryInfo.getWarnings();
+            }
+            else {
+                warnings = ImmutableList.of();
```

Comment:
Technically, `ImmutableList.of()` doesn't create a new list. OTOH maybe `warnings` shouldn't be nullable in `QueryResults`? We generally don't like nulls, i suppose.